src/cpp/rtps/persistence/SQLite3PersistenceService.cpp

```diff
@@ -284,6 +300,21 @@ bool SQLite3PersistenceService::load_writer_from_storage(
             change->serializedPayload.length = size;
             memcpy(change->serializedPayload.data, sqlite3_column_blob(load_writer_stmt_, 2), size);
 
+            // related sample identity
+            {
+                using namespace std;
+                // GUID_t
+                istringstream is(string(reinterpret_cast<const char*>(sqlite3_column_text(load_writer_stmt_, 3))));
+                auto& si = change->write_params.related_sample_identity();
+                is >> si.writer_guid();
+                // Sequence Number
+                SequenceNumber_t rsn(sqlite3_column_int64(load_writer_stmt_, 4));
+                si.sequence_number(rsn);
+            }
+
+            // timestamp
+            change->sourceTimestamp = Time_t(sqlite3_column_double(load_writer_stmt_, 5));
```

Comment:
Using `double` to store the sourcetimestap sure there will be loss of precision. I recomend to store the values of `rtps::Time_t`: seconds and nanoseconds.

Reply:
In order to avoid precision loss the *timestamps* are now stored as `int64_t`.